Mathlib/Data/EReal/Operations.lean

```diff
@@ -313,6 +315,17 @@ lemma neg_sub {x y : EReal} (h1 : x ≠ ⊥ ∨ y ≠ ⊥) (h2 : x ≠ ⊤ ∨ y
     - (x - y) = - x + y := by
   rw [sub_eq_add_neg, neg_add _ _, sub_eq_add_neg, neg_neg] <;> simp_all
 
+/-- Induction principle for `EReal`s splitting into cases `↑(x : ℝ≥0∞)` and `-↑(x : ℝ≥0∞)`.
+In the latter case, we additionally assume `0 < x`. -/
+@[elab_as_elim]
+def recENNReal {motive : EReal → Sort*} (coe : ∀ x : ℝ≥0∞, motive x)
+    (neg_coe : ∀ x : ℝ≥0∞, 0 < x → motive (-x)) (x : EReal) : motive x :=
```

Comment:
Can you prove `(x : ℝ≥0∞) : recENNReal x = coe x` and `{x : ℝ≥0∞} (hx : x ≠ 0) : recENNReal (-x) = negCoe x`?

Reply:
Proved 1 of them, got stuck with dependent type hell in the other one.

Reply:
Tried the second one, failed too